package.json

```diff
@@ -48,13 +48,17 @@
     "lint-staged": "13.1.0",
     "mocha": "10.2.0",
     "mocha-multi-reporters": "1.5.1",
+    "nock": "^13.3.0",
     "semantic-release": "19.0.5"
   },
   "lint-staged": {
     "*.js": "eslint",
     "*.cjs": "eslint"
   },
   "dependencies": {
+    "@adobe/fetch": "^3.3.1",
+    "async": "^3.2.4",
+    "fetch-retry": "^5.0.3",
```

Comment:
Shameless plug, but we have a recently open-sourced Adobe library that does something similar: [cloud-service-client](https://github.com/adobe/cloud-service-client). In addition to retrying, it can be configured to handle eventual consistency, cookies, and more. I don't think we're necessarily working with AEM here, but it was designed to solve many of the issues we've encountered when communicating with AEMaaCS.

Feel free to take a look if you'd like.